tests/testapp/tests/test_models.py

```diff
@@ -0,0 +1,57 @@
+import factory
+
+from django.test import TestCase
+from morango.models import DatabaseMaxCounter
+
+
+class DatabaseMaxCounterFactory(factory.DjangoModelFactory):
+
+    class Meta:
+        model = DatabaseMaxCounter
+
+
+class FilterMaxCounterTestCase(TestCase):
+
+    def setUp(self):
+        self.instance_a = "a" * 32
+        self.prefix_a = "AAA"
+        self.user_prefix_a = "AAA:user_id:joe"
+
+        self.instance_b = "b" * 32
+        self.prefix_b = "BBB"
+        self.user_prefix_b = "BBB:user_id:rick"
+        self.user2_prefix_b = "BBB:user_id:emily"
+
+        # instance A dmc
+        DatabaseMaxCounterFactory(instance_id=self.instance_a, partition=self.prefix_a, counter=15)
+        DatabaseMaxCounterFactory(instance_id=self.instance_a, partition=self.user_prefix_a, counter=20)
+        DatabaseMaxCounterFactory(instance_id=self.instance_a, partition=self.user2_prefix_b, counter=17)
+
+        # instance B dmc
+        DatabaseMaxCounterFactory(instance_id=self.instance_b, partition=self.user_prefix_a, counter=10)
+        DatabaseMaxCounterFactory(instance_id=self.instance_b, partition=self.prefix_b, counter=12)
+        DatabaseMaxCounterFactory(instance_id=self.instance_b, partition=self.user_prefix_b, counter=5)
+        DatabaseMaxCounterFactory(instance_id=self.instance_b, partition=self.user2_prefix_b, counter=2)
+
+    def test_filter_not_in_dmc(self):
+        fmcs = DatabaseMaxCounter.calculate_filter_max_counters("ZZZ")
+        self.assertFalse(fmcs)
+
+    def test_instances_for_one_partition_but_not_other(self):
+        fmcs = dict(DatabaseMaxCounter.calculate_filter_max_counters(self.user_prefix_a + "\n" + self.user_prefix_b))
```

Comment:
its returned as a list, so maybe just always coerce to dict but in the actual function call.